src/Microsoft.Diagnostics.Monitoring.StartupHook/ParameterCapturing/MethodSignature.cs

```diff
@@ -46,6 +46,10 @@ public static class Names
 
         public string MethodName { get; } = GetMethodName(method);
 
+        public int MethodToken { get; } = method.MetadataToken;
+
+        public Guid ModuleVersionId { get; } = method.Module.ModuleVersionId;
+
```

Comment:
I'm not 100% certain yet that this is actually safe-to-do here. These properties may throw on access for certain methods. I don't believe the parameter capturing infrastructure ever tries to access either of these properties except for when it goes to calculate ancillary boxing instructions which is guarded against this problem.

The rest of the infrastructure operates on function ids. It does exclude dynamic assemblies from being instrumented.

We either need to guard when accessing these, or validate that it is **always** safe to do given the parameter capturing infrastructure design. I'll think on it more, but it might be safest just to guard.

Reply:
Assuming we catch an exception, do we just want these 0'd out (in the case of MVID just an empty GUID), or should we specifically indicate that there was no value?

Reply:
Let's use 0'd out values as backups, this will be consistent with what we do for stack frames that don't have this data